lldb/include/lldb/Core/DemangledNameInfo.h

```diff
@@ -74,24 +74,48 @@ struct DemangledNameInfo {
     return BasenameRange.second > BasenameRange.first;
   }
 
+  /// Returns \c true if `BasenameRange` is empty.
+  bool isBasenameEmpty() const {
+    return BasenameRange.first == BasenameRange.second;
```

Comment:
Hmm I think I'd prefer having only a single API that tells us "do we have a valid entity or not". E.g., `hasValidQualifiers`/`hasValidBasename`/etc. Is there a reason we can't just modify the existing `hasXXX` helpers to be `Range.second >= Range.first`? If we are going to call `!hasXXX && !isEmpty` then might as well fold them into a single API.

But if there's reasons we can't do that, this seems fine too

Reply:
Thanks for adjusting. Mostly looks good now

Though I don't think `hasBasename` should allow for `start == end`. That's the one range that all `DemangledInfo` objects need to have to be considered valid. Could we keep that as `end > start`? I don't think that should break anything

Reply:
Switched to `>` instead of `>=` for `hasBasename`. This "fixed" the tests that I originally broke 👍